soc/xtensa/esp32/soc.h

```diff
@@ -31,6 +31,16 @@ static inline void esp32_clear_mask32(uint32_t v, uint32_t mem_addr)
 	sys_write32(sys_read32(mem_addr) & ~v, mem_addr);
 }
 
+static inline uint32_t esp32_core_id(void)
```

Comment:
AFAIK  `arch_proc_id()` depends on correct configuration of cpu_pointer structure, which only occurs for xtensa code that enables SMP, since is not  the case here, this function will always return the core 0, so I preferred to introduce a SoC specific function that returns the core id regardless the SMP or not.

I think this is a separate issue, correct the xtensa arch code in respect of this and after that submit a new PR to deprecate or remove this function in the future.

Reply:
Yes, this needs to be fixed at arch level.